### PR TITLE
Issue 6810 type annotate sqlalchemy.orm.events

### PR DESCRIPTION
**Pull request by g-kisenkov**
### Checklist

This pull request is:

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [ ] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

**Review by mlatysh — APPROVED** [review]

**Comment by g-kisenkov** [comment]
By te way, just wanted to add that this PR is sort-of half-finished as I couldn't figure out a better way to type out some of the places without using casts and `# type: ignore`'s. So there are going to be quite a few mypy errors. If some of you could suggest a solution to certain places, then feel free to open a change request, push commits or better open a stacked PR (https://blog.logrocket.com/using-stacked-pull-requests-in-github/).

**Comment by g-kisenkov** [comment]
@dzmitar
@mlatysh 

Ready for review.

File: lib/sqlalchemy/orm/events.py [hunk 1]
```diff
@@ -1396,12 +1481,16 @@ def my_before_commit(session):
 
     _dispatch_target = Session
 
-    def _lifecycle_event(fn):
+    def _lifecycle_event(  # type: ignore [misc]
```
Comment (dzmitar) [inline lib/sqlalchemy/orm/events.py:1484]: IMHO we can add @staticmethod and delete "# type: ignore [misc]".
Also, if "-> Callable[[SessionEvents, Session, Any], None]" we can delete "# type: ignore [return-value]"

Comment (dzmitar) [inline lib/sqlalchemy/orm/events.py:1484]: IMHO we can add @staticmethod and delete "# type: ignore [misc]".
Also, if "-> Callable[[SessionEvents, Session, Any], None]" we can delete "# type: ignore [return-value]"

**Comment by g-kisenkov** [comment]
Closing this issue as I've opened a PR into upstream repo (#9025).

**Pull request closed without merging**